io/native/src/main/scala/fs2/io/net/tls/S2nConnection.scala

```diff
@@ -51,7 +51,11 @@ private[tls] trait S2nConnection[F[_]] {
 
   def shutdown: F[Unit]
 
-  def applicationProtocol: F[String]
+  /** Returns [[None]] if there is no protocol being negotiated.
+    *
+    * @see [[https://aws.github.io/s2n-tls/doxygen/s2n_8h.html#ae53faa26669e258afff875d45140f14e]]
```

Comment:
The doc says:
> In the event of no protocol being negotiated, NULL is returned.

Is this actually equivalent to the empty string `""` case on the JVM? I'm not sure. On the JVM `None` is if it's not yet determined, but `""` is if it's not being used (not being negotiated?).

Reply:
I am even less qualified to answer this question, not familiar with TLS enough :)